src/SkeletalTrapezoidation.cpp

```diff
@@ -355,18 +355,18 @@ void SkeletalTrapezoidation::computeSegmentCellRange(vd_t::cell_type& cell, Poin
 
 SkeletalTrapezoidation::SkeletalTrapezoidation(const Polygons& polys, float transitioning_angle
     , coord_t discretization_step_size, coord_t transition_filter_dist, coord_t beading_propagation_transition_dist
-    ): polys(polys), transitioning_angle(transitioning_angle), discretization_step_size(discretization_step_size)
+    ): transitioning_angle(transitioning_angle), discretization_step_size(discretization_step_size)
         , transition_filter_dist(transition_filter_dist), beading_propagation_transition_dist(beading_propagation_transition_dist)
 {
-    init();
+    constructFromPolygons(polys);
```

Comment:
Nice solution, was thinking the same thing when I saw the comment.

Reply:
It was a comment from @Ghostkeeper originally, so I can't take full credit for that.